vumi/middleware/base.py

```diff
@@ -176,7 +177,19 @@ class MiddlewareStack(object):
     """
 
     def __init__(self, middlewares):
-        self.middlewares = middlewares
+        self.consume_middlewares = self._sort_by_priority(
+            middlewares, 'consume_priority')
+        self.publish_middlewares = self._sort_by_priority(
+            reversed(middlewares), 'publish_priority')
```

Comment:
I'm wondering whether we should make the two priorities work in ascending order instead of ascending for the one and ascending for the other, seems a bit surprising. It would mean breaking backwards compatibility I guess, though, but maybe that's okay,